corehq/util/workbook_json/excel.py

```diff
@@ -226,10 +217,19 @@ def _convert_float(value):
                 yield cell_values
         super(WorksheetJSONReader, self).__init__(iterator())
 
+    def row_count(self):
+        def parse_dimension(dimension):
+            import re
+            match = re.search(r'(\d+)', dimension)
+            if match:
+                return int(match.group(1))
+            return 0
+        return parse_dimension(self.worksheet.calculate_dimension())
```

Comment:
Why use an inner function here? What do you think of this instead?
```suggestion
        import re
        dimension = self.worksheet.calculate_dimension()
        match = re.search(r'(\d+)', dimension)
        return int(match.group(1)) if match else 0
```

Also, what does the result of `self.worksheet.calculate_dimension()` look like? Does it contain other text in addition to the number?

Reply:
Yeah I definitely meant to add tests around this function and agree the inner function isn't necessary. Thanks for the feedback.

Reply:
Ended up removing this entirely. See the conversation and commits below.